src/components/portfolio/DetailsMainBacktesting.tsx

```diff
@@ -0,0 +1,346 @@
+            yearBest: '2020',
+            growthRateBest: 10.5,
+          },
+          worstYear: {
+            changeWorst: -70.56,
+            yearWorst: '2019',
+            growthRateWorst: -5.6,
+          },
+          finalPortfolioBalance: 970.43,
+          CAGR: 5.42,
+          standardDeviation: 12.1,
+          sharpeRatio: 0.65,
+        };
+        setTimeout(
+          () => dispatch({ type: 'setBacktesting', payload: backtestingMock }),
+          2000
+        );
+      }
+    } catch (e) {
+      if (isMounted.current) {
+        dispatch({ type: 'setError', payload: e });
+      }
+    }
+  };
+
+  // eslint-disable-next-line @typescript-eslint/no-unused-vars
+  const fetch = async (from: Date, to: Date) => {
+    dispatch({ type: 'setError', payload: undefined });
+    try {
+      const backTestingResponse = await API.backtesting(token, id, from, to);
+      if (isMounted.current) {
+        dispatch({ type: 'setBacktesting', payload: backTestingResponse });
+      }
+    } catch (e) {
+      if (isMounted.current) {
+        dispatch({ type: 'setError', payload: e });
+      }
+    }
+  };
+
+  // on initial mount
+  useEffect(() => {
+    mockFetch(state.selectedFrom, state.selectedTo);
+    return () => {
+      isMounted.current = false;
+    };
+    // deps must be empty because the function should only be called on mount.
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+  }, []);
+
+  const onClickUpdate = () => {
+    // disable button until new result is being displayed
+    dispatch({ type: 'setUpdateDisabled', payload: true });
+    // remove error from last click
+    dispatch({ type: 'setValid', payload: true });
+    if (
+      state.selectedTo < state.selectedFrom ||
+      new Date() < state.selectedTo
+    ) {
+      dispatch({ type: 'setValid', payload: false });
+      dispatch({ type: 'setUpdateDisabled', payload: false });
+    } else {
+      dispatch({ type: 'setBacktesting', payload: undefined });
+      mockFetch(state.selectedFrom, state.selectedTo);
+      dispatch({ type: 'setUpdateDisabled', payload: false });
+    }
+  };
+
+  return (
+    <div className={classes.backtestingWrapper}>
+      <h2 className={classes.subtitle}>
+        {t('portfolio.details.backtesting.subheading')}
+      </h2>
+      <div className={classes.datePicker}>
+        <form className={classes.form} noValidate>
+          <span className={classes.subtitle}>
+            {t('portfolio.details.from')}:
+          </span>
+          <MuiPickersUtilsProvider utils={MomentUtils}>
+            <KeyboardDatePicker
+              InputProps={{ className: classes.datePickerInput }}
+              value={state.selectedFrom}
+              onChange={(date) =>
+                dispatch({
+                  type: 'setFrom',
+                  // if the onchange date is null don't change the state
+                  payload: date ? date.toDate() : state.selectedFrom,
+                })
+              }
```

Comment:
If the state should not be updated, there is no need to call `dispatch`.

```suggestion
              onChange={(date) => {
                // if the onchange date is null don't change the state
                if (date !== null) {
                  dispatch({
                    type: 'setFrom',
                    payload: date.toDate(),
                  });
                }
              }}
```